lib/views/View.js

```diff
@@ -0,0 +1,73 @@
+var _ = require('lodash');
+
+function View () {
+  this.initialize.apply(this, arguments);
+}
+
+View.prototype.initialize = function (module) {
+  this.moduleConfig = module.moduleConfig;
+  this.data = module.dataSource.data;
+  this.axes = module.axes;
+
+  this.marshalData();
+};
+
+View.prototype.marshalData = function () {
+  var yAxes = this.axes.y;
+
+  this.reverseData = false;
+  this.groupedData = false;
+
+  if (this.moduleConfig['module-type'] !== 'grouped_timeseries') {
+    //For a delta we're only interested in the first axis for everything but grouped_timeseries
+    yAxes = [yAxes[0]];
+  }
+
+  if (this.moduleConfig['data-source'] && this.moduleConfig['data-source']['query-params']) {
+
+    if (this.moduleConfig['data-source']['query-params'].period) {
+      this.reverseData = true;
+    }
+
+    if (this.moduleConfig['data-source']['query-params'].group_by && yAxes.length > 1) {
```

Comment:
Could break this and the next if.. clause out to separate functions to help readability?